pkg/util/nativeutils/rpcbus/rpcbus.go

```diff
@@ -130,19 +130,31 @@ func (bus *RPCBus) Call(t topics.Topic, req Request, timeOut time.Duration) (int
 }
 
 func (bus *RPCBus) callTimeout(reqChan chan<- Request, req Request, timeOut time.Duration) (interface{}, error) {
+	timer := time.NewTimer(timeOut)
+
 	select {
 	case reqChan <- req:
-	case <-time.After(timeOut):
+	case <-timer.C:
 		return bytes.Buffer{}, ErrRequestTimeout
 	}
 
+	if !timer.Stop() {
+		<-timer.C
```

Comment:
Should we drain this chan in a separate goroutine so we don't block `callTimeout`

Reply:
I don’t think that’s necessary, at most we are draining a single element.
If it becomes a performance bottleneck, we can always optimize later.

Reply:
To be honest, I couldn't find where in `timerproc` or `deltimerLocked` the timer callback (the one that writes to timer.C) is called but let's rely on the example from documentation.